hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/LegacyReplicationManager.java

```diff
@@ -1273,6 +1273,7 @@ private List<ContainerReplica> getReplicationSources(ContainerInfo container,
         // maintenance nodes, as the replicas will remain present in the
         // container manager, even when they go dead.
         .filter(r -> getNodeStatus(r.getDatanodeDetails()).isHealthy()
+            && !r.getDatanodeDetails().isDecomissioned()
```

Comment:
I don't think this is the correct place to filter out decommissioned nodes. This is forming a list of replication source nodes - it is valid to replicate from a decommissioning host, and in some cases it is necessary.

Reply:
The correct place to filter out the decommission / maintenance nodes would be in the method "replicateAnyWithTopology", as that is where the exclude list is formed to pass into the placement policy. But we would also need to take care of this in the new replication manager.

Reply:
Hi @sodonnel , thanks for the review. Handled for both Legacy and New replication manager.